(AAA) Access for Enterprises guide (#36600)
DataDog/documentation: content/en/getting_started/access_for_enterprises/assigning_users.md

---
title: Assigning Users to Roles and Teams
description: Learn the distinction between Roles and Teams and how to assign users at scale.
further_reading:
- link: "/getting_started/teams/"
  tag: "Documentation"
  text: "Getting Started with Teams"
- link: "/account_management/saml/"
  tag: "Documentation"
  text: "SAML Configuration"
- link: "/account_management/scim/"
  tag: "Documentation"
  text: "SCIM Provisioning"
- link: "/api/latest/teams/"
  tag: "Documentation"
  text: "Teams API Reference"
---

## Overview

With the [permission model and starter roles][1] defined, the next step is assigning users to the right Roles and Teams. This section covers the distinction between Roles and Teams for access control, how they work together, and practical guidance on assignment.

## Roles vs. Teams

> In Datadog, users can be mapped to both Roles and Teams. A user may have one or more Roles, and one or more Teams. Both can be targets of access policies. Understanding when to use each is important for building a maintainable access model.

Reviewer: ```suggestion
In Datadog, users can be assigned one or more roles and be added to one or more teams. Both can be used to scope permissions, but they serve different purposes.
```

> **Roles** define a user's coarse permissions: whether they can use a feature, and which feature capabilities they can access. For access, this could mean:

Reviewer: Is it necessary to use "course permissions"? It reads like a definition within the definition.

```suggestion
**Roles** define what features a user has access to and which feature capabilities they can access. For access, this could mean:
```

- Who should be able to manage other users' permissions in Datadog
- Who is never able to see any Logs data
- Who is able to view Monitors, but not edit any
- Who can see sensitive data within a given Team (for example, which Team members can see personally identifiable information or PII)

**Teams** define users who work in the same problem space, on the same technical or product areas (such as services or repos), or need similar Monitor alerts. For access, this could mean:

- Who is able to edit a set of Monitors, because they own the underlying infrastructure
- Who is able to see sensitive Logs that are related to that Team's work
- Who is able to see any data related to a sensitive service owned by that Team

> ### How Roles and Teams work together

Reviewer: ```suggestion
## How Roles and Teams work together
```

A practical model for most enterprises:

- Use **Roles** to define broad permission tiers (what a user *can do* across the platform).
- Use **Teams** to define ownership boundaries (what resources and data a user *is responsible for*).
- Where a specific Team needs elevated permissions, use **Team-linked Roles** (in Preview) to assign the role to the Team rather than to individual users. When you assign a role to a Team, each member receives the role.

## Admin override groups

> When you begin restricting resources to specific teams (covered in [Protecting Assets][2]), be aware that those restrictions apply to everyone, including administrators. If an admin is not a member of the owning team, they can be locked out of resources they need to manage.

Reviewer: Does this need to be introduced here? Seems like we're jumping ahead. Might be better to address this in just the Protecting Assets page, or in the Permissions page where it goes into the individual roles.

Author: excellent catch :) moved it to the Permissions page!

To avoid this, establish a break-glass **admin override team** (such as "Platform Admins" or "Datadog Governance") and include it in the access list for all critical resources. This helps ensure that your administrative staff always retains access for troubleshooting and governance, even as you tighten team-based restrictions.

**Note:** Users with the `user_access_manage` permission are able to grant themselves elevated access as a failsafe.

> ## Request Access (Preview)

Reviewer: Are there docs on this? If so, link out to this feature and remove the Preview label so you don't have to maintain this here.

Author: double-checked with the PM and a draft is in the works. gonna make a ticket for myself to come back and add the link when those docs are published (that, and the other in-preview features)

Datadog offers Request Access (in Preview) for situations where users need temporary or elevated access beyond their standard role, such as responding to an urgent incident or debugging a production issue in another team's domain. This allows end users to request access to resources they don't have, with approval workflows for administrators.

## Recommendations

- **Use Roles for what users *can do*, and Teams for what users *are responsible for*.** Don't conflate the two.
- **Establish an admin override team** before you begin restricting resources to specific teams.
- **Automate provisioning.** Map your IdP groups to Datadog Roles and Teams. Manual user management does not scale.
- **Audit role and team assignments periodically.** Look for users with more permissions than they need, roles with no users, and stale team memberships.

> 

Reviewer: Do you need to add Next Steps like the other pages?

```suggestion
## Next steps

Restrict who can edit or view Dashboards, Monitors, and other assets. View common patterns for [Protecting Assets][2].
```
